npm_and_yarn/lib/dependabot/npm_and_yarn.rb

```diff
@@ -26,6 +26,13 @@
   end
 )
 
+## A type used for defining a proc that creates a new error object
+NewErrorProc = T.type_alias do
+  T.proc
+   .params(message: String, error: Dependabot::DependabotError, params: T::Hash[Symbol, T.untyped])
```

Comment:
`params` parameter added to handle function in case where error handling uses parameters such as `dependecies`, `dependency_files`, or `yarn_lock`

Reply:
> `params` parameter added to handle function in case where error handling uses parameters such as `dependecies`, `dependency_files`, or `yarn_lock`